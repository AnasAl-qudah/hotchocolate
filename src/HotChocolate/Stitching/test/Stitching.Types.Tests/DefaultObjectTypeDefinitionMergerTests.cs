using HotChocolate.Stitching.Types.Bindings;
using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.Stitching.Types;

public class DefaultObjectTypeDefinitionMergerTests
{
    [Fact]
    public void Merge_Simple_Type()
    {
        // arrange
        ObjectTypeDefinition.TryParse(
            @"type Foo {
                a: String
            }",
            "Abc",
            out ObjectTypeDefinition? a);

        ObjectTypeDefinition.TryParse(
            @"type Foo {
                b: String
            }",
            "Def",
            out ObjectTypeDefinition? b);

        // act
        var merger = new DefaultObjectTypeDefinitionMerger();
        merger.MergeInto(a!, b!);

        // assert
        Assert.Collection(
            b!.Bindings,
            t => Assert.Equal("Def", Assert.IsType<SourceBinding>(t).Source),
            t => Assert.Equal("Abc", Assert.IsType<SourceBinding>(t).Source));
        b.Definition.ToString().MatchSnapshot();
    }
}
