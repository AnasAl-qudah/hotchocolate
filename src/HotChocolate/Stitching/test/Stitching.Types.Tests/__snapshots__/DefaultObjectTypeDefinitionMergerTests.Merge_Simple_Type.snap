{
  "Kind": "ObjectTypeDefinition",
  "Description": null,
  "Interfaces": [],
  "Fields": [
    {
      "Kind": "FieldDefinition",
      "Description": null,
      "Arguments": [],
      "Type": {
        "Kind": "NamedType",
        "Location": {
          "Start": 30,
          "End": 50,
          "Line": 2,
          "Column": 20
        },
        "Name": {
          "Kind": "Name",
          "Location": {
            "Start": 30,
            "End": 50,
            "Line": 2,
            "Column": 20
          },
          "Value": "String"
        }
      },
      "Location": {
        "Start": 27,
        "End": 50,
        "Line": 2,
        "Column": 17
      },
      "Name": {
        "Kind": "Name",
        "Location": {
          "Start": 27,
          "End": 29,
          "Line": 2,
          "Column": 17
        },
        "Value": "b"
      },
      "Directives": []
    },
    {
      "Kind": "FieldDefinition",
      "Description": null,
      "Arguments": [],
      "Type": {
        "Kind": "NamedType",
        "Location": {
          "Start": 30,
          "End": 50,
          "Line": 2,
          "Column": 20
        },
        "Name": {
          "Kind": "Name",
          "Location": {
            "Start": 30,
            "End": 50,
            "Line": 2,
            "Column": 20
          },
          "Value": "String"
        }
      },
      "Location": {
        "Start": 27,
        "End": 50,
        "Line": 2,
        "Column": 17
      },
      "Name": {
        "Kind": "Name",
        "Location": {
          "Start": 27,
          "End": 29,
          "Line": 2,
          "Column": 17
        },
        "Value": "a"
      },
      "Directives": []
    }
  ],
  "Location": {
    "Start": 0,
    "End": 50,
    "Line": 1,
    "Column": 1
  },
  "Name": {
    "Kind": "Name",
    "Location": {
      "Start": 5,
      "End": 10,
      "Line": 1,
      "Column": 6
    },
    "Value": "Foo"
  },
  "Directives": []
}
